services/wallet/transfer/controller.go

```diff
@@ -192,6 +192,30 @@ func (c *Controller) LoadTransferByHash(ctx context.Context, rpcClient *rpc.Clie
 	return nil
 }
 
+func getFixedTransferType(tx Transfer) Type {
+	// erc721 transfers share signature with erc20 ones, so they are both (cached and new)
+	// categorized as erc20 by the Downloader. We fix this on the fly for the moment, until
+	// the Downloader gets refactored.
+	if tx.Type == erc20Transfer {
+		switch len(tx.Log.Topics) {
+		case erc20TransferEventIndexedParameters:
+			// do nothing
+		case erc721TransferEventIndexedParameters:
+			return erc721Transfer
+		default:
+			return unknownTokenTransfer
+		}
+	}
+	return tx.Type
+}
+
+func transfersToTransferViews(transfers []Transfer) []View {
+	for idx, tx := range transfers {
+		transfers[idx].Type = getFixedTransferType(tx)
+	}
+	return castToTransferViews(transfers)
```

Comment:
Do you mean the `case` statement in `CastToTransferView` from `view.go`? So, my reasoning was:

-) Ideally this would be fixed at the `downloader.go` level (properly set the type to the `Transfer` object, and fix the ones we've got stored in the DB).
-) Next best option is to go one level up, which is `controller.go`
-) `view.go` seems to be "too close to the top layer" for being in charge of this.

Reply:
I mean a cast function *